src/fft.jl

```diff
@@ -0,0 +1,576 @@
+                            (overlap_y .- first(dst_domain.indexes[2]) .+ 1),
+                            (overlap_z .- first(dst_domain.indexes[3]) .+ 1))
+                        
+                        tag = src_idx * nranks + dst_idx
+                        req = MPI.Irecv!(recv_buf, src_chunk.handle.rank, tag, comm)
+                        push!(reqs, req)
+                    end
+                end
+            end
+        end
+    end
+    
+    # Second pass: Process local data and initiate sends
+    for (src_idx, src_chunk) in enumerate(src.chunks)
+        if src_chunk.handle.rank == rank
+            src_data = fetch(src_chunk)
+            src_domain = src.subdomains[src_idx]
+            
+            for (dst_idx, dst_chunk) in enumerate(dest.chunks)
+                dst_domain = dest.subdomains[dst_idx]
+                
+                overlap_x = intersect(src_domain.indexes[1], dst_domain.indexes[1])
+                overlap_y = intersect(src_domain.indexes[2], dst_domain.indexes[2])
+                overlap_z = intersect(src_domain.indexes[3], dst_domain.indexes[3])
+                
+                if !isempty(overlap_x) && !isempty(overlap_y) && !isempty(overlap_z)
+                    if dst_chunk.handle.rank == rank
+                        # Local copy
+                        dst_data = fetch(dst_chunk)
+                        src_indices = (
+                            overlap_x .- first(src_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(src_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(src_domain.indexes[3]) .+ 1
+                        )
+                        dst_indices = (
+                            overlap_x .- first(dst_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(dst_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(dst_domain.indexes[3]) .+ 1
+                        )
+                        dst_data[dst_indices...] .= view(src_data, src_indices...)
+                    else
+                        # Remote send using views!!!
+                        src_indices = (
+                            overlap_x .- first(src_domain.indexes[1]) .+ 1,
+                            overlap_y .- first(src_domain.indexes[2]) .+ 1,
+                            overlap_z .- first(src_domain.indexes[3]) .+ 1
+                        )
+                        send_data = view(src_data, src_indices...)
+                        
+                        tag = src_idx * nranks + dst_idx
+                        req = MPI.Isend(send_data, dst_chunk.handle.rank, tag, comm)
+                        push!(reqs, req)
+                    end
+                end
+            end
+        end
+    end
+    
+    if !isempty(reqs)
+        MPI.Waitall(reqs)
+    end
+    
+    MPI.Barrier(comm)
+end
+
+
+function fft(A::DArray{T,3}, B::DArray{T,3}, C::DArray{T,3}, transforms, dims) where T
```

Comment:
This signature should probably be `AbstractFFTs.fft!(out::DArray{T,3}, A::DArray{T,3}; dims) where T`, and we can also introduce `AbstractFFTs.fft(A::DArray{T,3}; dims) where T`. This means:
- The `fft(A)` function will automatically allocate `out` and call `fft!(out, A; dims=dims)`
- The `fft!(out, A)` function will automatically allocate any intermediary `DArray`s for the user
- The user won't specify `transforms`, because this isn't a standard part of the `AbstractFFTs` API
- We're adding methods to `AbstractFFTs` so that existing code which calls `fft` or `fft!` will automatically be able to hit our `DArray` methods.

Reply:
```suggestion
function AbstractFFTs.fft(A::DArray{T,3}; dims, decomp::Decomposition=Slab()) where T
```